decidim-core/lib/decidim/core/test/shared_examples/comments_examples.rb

```diff
@@ -162,51 +149,67 @@ def visit_commentable_path
+        before do
+          visit resource_path
 
-        expect(page).to have_selector(".add-comment form")
+          expect(page).to have_selector(".add-comment form")
+        end
 
-        page.find(".opinion-toggle--ok").click
+        it "works according to the setting in the commentable" do
+          if commentable.comments_have_alignment?
+            page.find(".opinion-toggle--ok").click
 
-        within ".add-comment form" do
-          fill_in "add-comment-#{commentable.commentable_type}-#{commentable.id}", with: "I am in favor about this!"
-          click_button "Send"
-        end
+            within ".add-comment form" do
+              fill_in "add-comment-#{commentable.commentable_type}-#{commentable.id}", with: "I am in favor about this!"
+              click_button "Send"
+            end
 
-        within "#comments" do
-          expect(page).to have_selector "span.success.label", text: "In favor"
+            within "#comments" do
+              expect(page).to have_selector "span.success.label", text: "In favor"
+            end
+          else
+            expect(page).not_to have_selector(".opinion-toggle--ok")
+          end
         end
       end
     end
 
-    context "when votable option is enabled" do
+    describe "votable option" do
       before do
-        expect_any_instance_of(commentable.class).to receive(:comments_have_votes?).and_return(true)
+        visit resource_path
       end
 
-      it "user can upvote a comment" do
-        visit_commentable_path
-
-        within "#comment_#{comments[0].id}" do
-          expect(page).to have_selector(".comment__votes--up", text: /0/)
-          page.find(".comment__votes--up").click
-          expect(page).to have_selector(".comment__votes--up", text: /1/)
+      context "upvoting a comment" do
+        it "works according to the setting in the commentable" do
+          within "#comment_#{comments[0].id}" do
+            if commentable.comments_have_votes?
+              expect(page).to have_selector(".comment__votes--up", text: /0/)
+              page.find(".comment__votes--up").click
+              expect(page).to have_selector(".comment__votes--up", text: /1/)
+            else
+              expect(page).to_not have_selector(".comment__votes--up", text: /0/)
+            end
+          end
         end
       end
 
-      it "user can downvote a comment" do
-        visit_commentable_path
+      context "downvoting a comment" do
+        before do
+          visit resource_path
+        end
 
-        within "#comment_#{comments[0].id}" do
-          expect(page).to have_selector(".comment__votes--down", text: /0/)
-          page.find(".comment__votes--down").click
-          expect(page).to have_selector(".comment__votes--down", text: /1/)
+        it "works according to the setting in the commentable" do
+          within "#comment_#{comments[0].id}" do
+            if commentable.comments_have_votes?
```

Comment:
> How do you know which branch of the if is running?

To further clarify this. We can't easily know at the moment. That's part of the reason why your suggestion is a good one...

I checked though and we are only testing the negative case against the "dummy feature". But previously we were never testing it!

```shell
$ ag 'include_examples "comments' --no-numbers
```
```ruby
decidim-comments/spec/features/comments_spec.rb
  include_examples "comments"

decidim-budgets/spec/features/comments_spec.rb
  include_examples "comments"

decidim-proposals/spec/features/comments_spec.rb
  include_examples "comments"

decidim-results/spec/features/comments_spec.rb
  include_examples "comments"

decidim-pages/spec/features/comments_spec.rb
  include_examples "comments"
```

```shell
$ ag "def comments_have_votes?" -A2 --no-numbers
```

```ruby
decidim-comments/lib/decidim/comments/commentable.rb
        def comments_have_votes?
          false
        end

decidim-budgets/app/models/decidim/budgets/project.rb
      def comments_have_votes?
        true
      end

decidim-proposals/app/models/decidim/proposals/proposal.rb
      def comments_have_votes?
        true
      end

decidim-results/app/models/decidim/results/result.rb
      def comments_have_votes?
        true
      end

decidim-pages/app/models/decidim/pages/page.rb
      def comments_have_votes?
        true
      end
```